### PR TITLE
Dropout: rescale data at training time (breaks bw compat.)

### PR DESCRIPTION
**Pull request by mmmikael**
Following the discussion https://groups.google.com/forum/#!topic/keras-users/PYbBnZXCvO4, here is a simple change to the '''Dropout''' layer that follows the Caffe implementation. 

Data rescaling is done during training instead of testing. This has the advantage of making the layer equal to identity at test time. Therefore, it can be safely removed in a production environment.

However, this will break compatibility with previously saved models.

### CODE REVIEW

**Comment by fchollet** [comment]
As discussed, I believe this is the right choice.
